app/renderer/actions/Inspector.js

```diff
@@ -510,6 +512,107 @@ export function setLocatorTestElement (elementId) {
+  const UPPER_FILTER_LIMIT = 10;
+
+  // Parse the source tree and find all nodes whose bounds match the expected bounds
+  // Return the path + xpath of each node
+  function findPathsMatchingBounds () {
+    if (bounds && source.children && source.children[0].attributes) {
+      if (source.children[0].attributes.bounds) {
+        const [endX, endY] = [bounds.location.x + bounds.size.width, bounds.location.y + bounds.size.height];
+        const coords = `[${bounds.location.x},${bounds.location.y}][${endX},${endY}]`;
+        return findPathsFromCoords(source.children, coords);
+      } else if (source.children[0].attributes.x) {
+        const combinedBounds = {
+          x: String(bounds.location.x),
+          y: String(bounds.location.y),
+          height: String(bounds.size.height),
+          width: String(bounds.size.width),
+        };
+        return findPathsFromBounds(source.children, combinedBounds);
+      }
+    }
+    return null;
+  }
+
+  // Recursive function for parsing source tree when elements have 'bounds' property
+  function findPathsFromCoords (trees, coords) {
+    let collectedPaths = [];
+    for (const tree of trees) {
+      if (tree.attributes.bounds === coords) {
+        collectedPaths.push([tree.path, tree.xpath]);
+      }
+      if (tree.children.length) {
+        collectedPaths.push(...findPathsFromCoords(tree.children, coords));
+      }
+    }
+    return collectedPaths;
+  }
+
+  // Recursive function for parsing source tree when elements have 'x/y/height/width' properties
+  function findPathsFromBounds (trees, bounds) {
+    let collectedPaths = [];
+    for (const tree of trees) {
+      if (tree.attributes.x === bounds.x && tree.attributes.y === bounds.y
+        && tree.attributes.height === bounds.height && tree.attributes.width === bounds.width) {
+        collectedPaths.push([tree.path, tree.xpath]);
+      }
+      if (tree.children.length) {
+        collectedPaths.push(...findPathsFromBounds(tree.children, bounds));
+      }
+    }
+    return collectedPaths;
+  }
+
+  // If findPathsMatchingBounds found multiple items,
+  // use Appium findElement to filter further by element ID
+  async function filterFoundPaths (foundPaths, dispatch, getState) {
+    if (foundPaths) {
+      if (foundPaths.length === 1) {
+        return foundPaths[0][0];
+      } else if (foundPaths.length !== 0 && foundPaths.length <= UPPER_FILTER_LIMIT) {
+        return await findElementWithMatchingId(foundPaths, dispatch, getState);
+      }
+    }
+    return null;
+  }
+
+  // Calls Appium findElement for each provided xpath, and returns the path
+  // of the element whose ID matches the expected ID
+  async function findElementWithMatchingId (foundPaths, dispatch, getState) {
+    for (const path of foundPaths) {
+      const action = callClientMethod({strategy: 'xpath', selector: path[1]});
+      const { el } = await action(dispatch, getState);
+      if (el && el.elementId === id) {
```

Comment:
could it be `el?.elementId` ?

Reply:
i don't think we have any guarantee that the same element found by multiple invocations of `findElement` will have the same ID. the drivers do not make any such guarantee, anyway, i'm pretty sure.

Reply:
@mykola-mokhnach unfortunately we can't use optional chaining yet. The app is built with `parcel` v1, which depends on `terser` 3.x, but optional chaining support was only added in `terser` 5.2.0.

Reply:
@jlipps I agree, I've definitely seen common ID changes with iOS. But if this happens, the user will simply receive the error message. I'd say this is better than trying to match using other element properties that are less unique, which may result in selecting the incorrect node.